frontend/src/pages/projects/projectSharing/ProjectSharingTable.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import Table from '~/components/Table';
+import { RoleBindingKind } from '~/k8sTypes';
+import { deleteRoleBinding, generateRoleBindingProjectSharing, createRoleBinding } from '~/api';
+import ProjectSharingTableRow from './ProjectSharingTableRow';
+import { columnsProjectSharingUser, columnsProjectSharingGroup } from './data';
+import { ProjectSharingRBType } from './types';
+
+type ProjectSharingTableProps = {
+  type: ProjectSharingRBType;
+  permissions: RoleBindingKind[];
+  typeAhead: string[];
+  onCancel: () => void;
+  onError: (error: Error) => void;
+  refresh: () => void;
+};
+
+const ProjectSharingTable: React.FC<ProjectSharingTableProps> = ({
+  type,
+  permissions,
+  typeAhead,
+  onCancel,
+  onError,
+  refresh,
+}) => {
+  const [editCell, setEditCell] = React.useState<string[]>([]);
+
+  return (
+    <Table
+      variant="compact"
+      data={permissions}
+      columns={
+        type === ProjectSharingRBType.USER ? columnsProjectSharingUser : columnsProjectSharingGroup
+      }
+      disableRowRenderSupport
+      rowRenderer={(rb) => (
+        <ProjectSharingTableRow
+          key={rb.metadata?.name || ''}
+          obj={rb}
+          isEditing={rb.subjects[0]?.name === '' || editCell.includes(rb.metadata.name)}
+          typeAhead={typeAhead}
+          onCreateOrEditRoleBinding={(name, roleType, obj) => {
```

Comment:
```suggestion
          onRoleBindingChange={(name, roleType) => {
```

Small nit: Avoid `And` and `Or` in function callbacks -- makes it really wordy.

Also... The `rb` is available in closure scope -- make it easier on the signature.

Reply:
Mmmmm, i wasn't sure about that, that's why I added it in the parameter, if it's fine using the scope object ill do that.